relation_engine_server/api_versions/api_v1.py

```diff
@@ -15,25 +14,38 @@
 api_v1 = flask.Blueprint('api_v1', __name__)
 
 
-@api_v1.route("/data_sources", methods=["GET"])
+@api_v1.route("/data_sources", methods=['GET'])
 def list_data_sources():
-    data_sources = load_data_sources.list_all()
+    # note the custom response format is used by the frontend, so this endpoint is provided
+    # in addition to the /specs/data_sources endpoint
+
+    data_sources = spec_loader.get_names('data_sources')
     return flask.jsonify({'data_sources': data_sources})
 
 
-@api_v1.route("/data_sources/<name>", methods=["GET"])
-def show_data_source(name):
-    data_source = load_data_sources.fetch_one(name)
+@api_v1.route("/data_sources/<name>", methods=['GET'])
+def fetch_data_source(name):
+
+    data_source = spec_loader.get_schema('data_source', name)
     return flask.jsonify({'data_source': data_source})
 
 
+@api_v1.route('/specs/data_sources', methods=['GET'])
+def show_data_sources():
+    """Show the current data sources loaded from the spec."""
+    name = flask.request.args.get('name')
+    if name:
+        return flask.jsonify(spec_loader.get_schema('data_source', name))
+    return flask.jsonify(spec_loader.get_names('data_sources'))
+
+
 @api_v1.route('/specs/stored_queries', methods=['GET'])
 def show_stored_queries():
     """Show the current stored query names loaded from the spec."""
     name = flask.request.args.get('name')
     if name:
-        return {'stored_query': spec_loader.get_stored_query(name)}
-    return flask.jsonify(spec_loader.get_stored_query_names())
+        return flask.jsonify({'stored_query': spec_loader.get_schema('stored_query', name)})
```

Comment:
this endpoint wraps the response in an extra `dict`, unlike the `collections` endpoint. I will add an issue to standardise the response.